opera/rules_test.go

```diff
@@ -0,0 +1,67 @@
+package opera
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFeatureSet_CanBeConvertedToString(t *testing.T) {
+
+	tests := map[string]FeatureSet{
+		"sonic":   SonicFeatures,
+		"allegro": AllegroFeatures,
+		"unknown": FeatureSet(math.MaxInt),
+	}
+
+	for expected, fs := range tests {
+		if fs.String() != expected {
+			t.Errorf("Expected %s, got %s", expected, fs.String())
+		}
+	}
+}
+
+func TestFeatureSet_CanBeConvertedToUpgrades(t *testing.T) {
+
+	tests := map[FeatureSet]struct {
+		expectedUpgrades Upgrades
+		expectedErr      string
+	}{
+		SonicFeatures: {
+			expectedUpgrades: Upgrades{
+				Berlin:  true,
+				London:  true,
+				Llr:     false,
+				Sonic:   true,
+				Allegro: false,
+			},
+		},
+		AllegroFeatures: {
+			expectedUpgrades: Upgrades{
+				Berlin:  true,
+				London:  true,
+				Llr:     false,
+				Sonic:   true,
+				Allegro: true,
+			},
+		},
```

Comment:
```suggestion
		SonicFeatures: {
			expectedUpgrades: GetSonicUpgrades(),
		},
		AllegroFeatures: {
			expectedUpgrades: GetAllegroUpgrades(),
		},
```
do you think doing it like this would hurt readability ?

Reply:
this does not test correctness of the function